api/client/service.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package nighthawk.client;
+
+import "api/client/options.proto";
+import "api/client/output.proto";
+
+message StartRequest {
+  CommandLineOptions options = 1;
+}
+
+message UpdateRequest {
+}
+
+message ExecutionRequest {
+  oneof command_specific_options {
+    StartRequest start_request = 1;
+    UpdateRequest update_request = 2;
+  }
+}
+
+message ExecutionResponse {
+  Output output = 1;
+}
+
+service NighthawkService {
+
+  /*
+  Accepts a ExecutionRequest, which can be used to initiate, update, or cancel a test.
+  Returns a stream of ExecutionResponse objects which will contain results for all successfull
```

Comment:
What is returned when an execution fails?

Reply:
Changed the proto a bit & updated the docs.
Earlier, we would only support one full process context run per connection.
Passing errors in the grpc status trailers would suffice.
Now we support multiple back-to-back runs, and indicate success plus a message on each response we emit. 
If any response indicated failure, we return a a failure status at the grpc-level as well.
Let me know what you think.